### PR TITLE
Support native profiling in benchmark_hat_package

### PR DESCRIPTION
**Pull request by CaptnJackSparrow**
We can invoke native profiling as in the below example:

`python3 -m hatlib.benchmark_hat_package /home/ritdas/onnx2exe/node_output_acc/layer_normalization_BS_SL_256_pkg.hat --cpp --min_time_in_sec 1`

and this would produce something like:

```
                                    function_name       mean           median_of_means  mean_of_small_means  robust_mean              min_of_means
0   layer_normalization_BS_SL_256_1_10_256_b007703... 0.00000002             [1.52716e-08]           0.00000001   0.00000002  [1.0001333333333332e-08]
1   layer_normalization_BS_SL_256_1_112_256_c5afd4... 0.00000098   [9.425019801980199e-07]           0.00000086   0.00000095   [7.313217821782179e-07]
2   layer_normalization_BS_SL_256_1_11_256_4a9f8b6... 0.00000002  [1.9885022692889557e-08]           0.00000002   0.00000002  [1.1886535552193646e-08]
3   layer_normalization_BS_SL_256_1_120_256_e3f425... 0.00000086   [8.516620370370372e-07]           0.00000079   0.00000085   [7.321574074074073e-07]
4   layer_normalization_BS_SL_256_1_133_256_d5df46... 0.00000112  [1.0595863157894737e-06]           0.00000100   0.00000107   [9.212284210526314e-07]
5   layer_normalization_BS_SL_256_1_18_256_a17c09b... 0.00000003  [2.4674587458745873e-08]           0.00000002   0.00000003  [2.0502970297029704e-08]
6   layer_normalization_BS_SL_256_1_192_256_915ee0... 0.00000217   [2.138241176470588e-06]           0.00000203   0.00000214  [1.8757220588235293e-06]

```

### CODE REVIEW

File: hatlib/hat_package.py [hunk 1]
```diff
@@ -181,40 +178,39 @@ def hat_package_to_func_dict(hat_pkg: HATPackage) -> AttributeDict:
     hat_dir_path, _ = os.path.split(hat_pkg.hat_file_path)
 
     for func_name, func_desc in hat_pkg.hat_file.function_map.items():
-
         launches = func_desc.launches
-        if not launches and shared_lib:
-
+        if not enable_native_profiling and not launches and shared_lib:
```
Comment (lisaong) [inline hatlib/hat_package.py:182]: maybe `_make_cpu_func` should be renamed to `_make_python_cpu_func`, because enable_native_profiling still profiles CPU functions, but natively.

File: hatlib/host_loader.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+#ifdef _MSC_VER
+#define DLL_EXPORT extern "C" __declspec( dllexport )
+#else
+#define DLL_EXPORT extern "C"
+#endif
+
+DLL_EXPORT void timer({intput_args_decl}, double* timing)
+{
+    auto start = std::chrono::high_resolution_clock::now();
+    {func_to_profile}
```
Comment (lisaong) [inline hatlib/host_loader.py:26]: this can be punted to another PR, but some functions, such as range() will need a way for the caller to filter inputs or provide a Python generator function to generate valid inputs.

**Review by lisaong — APPROVED** [review]

**Comment by lisaong** [comment]
> We can invoke native profiling as in the below example:
> 
> `python3 -m hatlib.benchmark_hat_package /home/ritdas/onnx2exe/node_output_acc/layer_normalization_BS_SL_256_pkg.hat --cpp --min_time_in_sec 1`
> 
> and this would produce something like:
> 
> ```
>                                     function_name       mean           median_of_means  mean_of_small_means  robust_mean              min_of_means
> 0   layer_normalization_BS_SL_256_1_10_256_b007703... 0.00000002             [1.52716e-08]           0.00000001   0.00000002  [1.0001333333333332e-08]
> 1   layer_normalization_BS_SL_256_1_112_256_c5afd4... 0.00000098   [9.425019801980199e-07]           0.00000086   0.00000095   [7.313217821782179e-07]
> 2   layer_normalization_BS_SL_256_1_11_256_4a9f8b6... 0.00000002  [1.9885022692889557e-08]           0.00000002   0.00000002  [1.1886535552193646e-08]
> 3   layer_normalization_BS_SL_256_1_120_256_e3f425... 0.00000086   [8.516620370370372e-07]           0.00000079   0.00000085   [7.321574074074073e-07]
> 4   layer_normalization_BS_SL_256_1_133_256_d5df46... 0.00000112  [1.0595863157894737e-06]           0.00000100   0.00000107   [9.212284210526314e-07]
> 5   layer_normalization_BS_SL_256_1_18_256_a17c09b... 0.00000003  [2.4674587458745873e-08]           0.00000002   0.00000003  [2.0502970297029704e-08]
> 6   layer_normalization_BS_SL_256_1_192_256_915ee0... 0.00000217   [2.138241176470588e-06]           0.00000203   0.00000214  [1.8757220588235293e-06]
> ```

I wonder if we should add a way to return time units in milliseconds rather than seconds (future work).

File: hatlib/host_loader.py (hunk 2 above)
Comment (CaptnJackSparrow) [reply hatlib/host_loader.py:26]: Yes, I faced the same issue with gather and its "indices" array.

Comment (CaptnJackSparrow) [reply hatlib/host_loader.py:26]: separate PR
